keras/engine/base_layer.py

```diff
@@ -226,7 +226,8 @@ def add_weight(self,
                    initializer=None,
                    regularizer=None,
                    trainable=True,
-                   constraint=None):
+                   constraint=None,
```

Comment:
As mentioned, add backend check here and only add weight and pass sparse argument for mxnet backend.